src/server/game/Spells/Auras/SpellAuraEffects.cpp

```diff
@@ -6000,12 +6047,21 @@ void AuraEffect::HandlePeriodicDamageAurasTick(Unit* target, Unit* caster) const
     if (crit)
         damage = caster->SpellCriticalDamageBonus(m_spellInfo, damage, target);
 
+    if (partial && m_period > 0)
+    {
+        int32 partialTickDuration = GetBase()->GetTickRemainingPeriod();
+        float dmgPercent = partialTickDuration / m_period;
```

Comment:
i'm basically rewriting the code of something i did for 4.3.4 server. It needs some testing, ofc...